src/app/contacts/edit-contact/edit-contact.component.ts

```diff
@@ -0,0 +1,66 @@
+import { Component } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+import { ContactsService } from 'src/app/contacts.service';
+import { Contact } from '../contacts';
+
+@Component({
+  selector: 'app-edit-contact',
+  templateUrl: './edit-contact.component.html',
+  styleUrls: ['./edit-contact.component.css'],
+})
+export class EditContactComponent {
+  // @ts-ignore
```

Comment:
The use of // @ts-ignore to ignore the form property type issue is not recommended. It's better to initialize your form in the constructor or declare it as possibly undefined. If you're sure form will be initialized in ngOnInit, you can use the non-null assertion operator when accessing it, but it's safer to check for its existence